jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -635,6 +635,7 @@ public void close() throws IOException
         {
             this.channel = channel;
             this.attachment = attachment;
+            _selectorManager.onAccepting(channel);
```

Comment:
The `Accept` constructor is called from `accept()` methods in `ManagedSelector` and `SelectorManager`, which in turn are called by either user code (on the client) or by the acceptor thread (on the server), which means that on the server we are calling user code that may block from the acceptor thread. Not sure I like it, but I don't see an easy way out.

Reply:
Indeed... and I don't even like calling out from a constructor.
I think accept listeners are very low level and we should document that they can be called from acceptor threads.